pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java

```diff
@@ -99,12 +103,22 @@ abstract class BaseInstanceSelector implements InstanceSelector {
   private volatile SegmentStates _segmentStates;
 
   BaseInstanceSelector(String tableNameWithType, ZkHelixPropertyStore<ZNRecord> propertyStore,
-      BrokerMetrics brokerMetrics, @Nullable AdaptiveServerSelector adaptiveServerSelector, Clock clock) {
+      BrokerMetrics brokerMetrics, @Nullable AdaptiveServerSelector adaptiveServerSelector, Clock clock,
+      boolean useFixedReplica) {
     _tableNameWithType = tableNameWithType;
     _propertyStore = propertyStore;
     _brokerMetrics = brokerMetrics;
     _adaptiveServerSelector = adaptiveServerSelector;
     _clock = clock;
+    _useFixedReplica = useFixedReplica;
+    // Using raw table name to ensure queries spanning across REALTIME and OFFLINE tables are routed to the same
+    // instance
+    _tableNameHashForFixedReplicaRouting = TableNameBuilder.extractRawTableName(tableNameWithType).hashCode();
```

Comment:
(minor) We can keep a positive hash here, which avoids the per query abs

Reply:
Ack